index.bs

```diff
@@ -989,36 +969,34 @@ A {{Point2D}} represents a location in a two dimensional space. The origin of co
     &lt;html>
     &lt;body>
     &lt;canvas>&lt;/canvas>
-    &lt;button onclick="stopGrabFrame()">Stop frame grab&lt;/button>
+    &lt;button id="stopButton">Stop frame grab&lt;/button>
     &lt;script>
-      const canvas = document.querySelector('canvas');
-
-      var interval;
-      var track;
-
-      navigator.mediaDevices.getUserMedia({video: true})
-        .then(gotMedia)
-        .catch(err => console.error('getUserMedia() failed: ', err));
-
-      function gotMedia(mediastream) {
-        track = mediastream.getVideoTracks()[0];
-        var imageCapture = new ImageCapture(track);
-        interval = setInterval(function () {
-          imageCapture.grabFrame()
-            .then(processFrame)
-            .catch(err => console.error('grabFrame() failed: ', err));
-        }, 1000);
-      }
-
-      function processFrame(imgData) {
-        canvas.width = imgData.width;
-        canvas.height = imgData.height;
-        canvas.getContext('2d').drawImage(imgData, 0, 0);
-      }
-
-      function stopGrabFrame(e) {
-        clearInterval(interval);
-        track.stop();
+      async function grabFrames() {
+        try {
+          const canvas = document.querySelector('canvas');
+          const video = document.querySelector('video');
+
+          const stream = await navigator.mediaDevices.getUserMedia({video: true});
+          video.srcObject = stream;
+          const [track] = stream.getVideoTracks();
+          try {
+            const imageCapture = new ImageCapture(track);
+
+            stopButton.onclick = () => track.stop();
```

Comment:
Shouldn't this stopButton variable be defined before use?

Reply:
`id="stopButton"` is enough in this case.

Reply:
Oh, I did not know that.